src/libmongoc/doc/mongoc_client_session_with_transaction.rst

```diff
@@ -22,6 +22,8 @@ This method has an internal time limit of 120 seconds, and will retry until that
 
 ``cb`` should not attempt to start new transactions, but should simply run operations meant to be contained within a transaction. The ``cb`` does not need to commit transactions; this is handled by the :symbol:`mongoc_client_session_with_transaction`. If ``cb`` does commit or abort a transaction, however, this method will return without taking further action.
 
+If a command inside ``cb`` fails, it may cause the transaction on the server to be aborted. This situation is normally handled transparently by the driver. However, if the application does not return that error from ``cb``, the driver will not be able to determine whether the transaction was aborted or not. The driver will then retry ``cb`` indefinitely. To avoid this situation, the application MUST NOT silently handle errors within ``cb``. If the application needs to handle errors within ``cb``, it MUST return them after doing so.
```

Comment:
https://github.com/mongodb/specifications/commit/13117d601fb6ea177e485880175b5be1651a7e46#diff-7307d020b76e062a84035a29df554e99f58f17401e60c9ac9448e919dbd0045eR985 also includes documentation about transaction behavior on command errors:

> Drivers MUST document that command errors inside a transaction may abort
the transaction on the server. An attempt to commit such transaction will be
rejected with ``NoSuchTransaction`` error.

Suggest adding that documentation to `mongoc_client_session_start_transaction.rst`